airflow/models/dagcode.py

```diff
@@ -147,7 +148,7 @@ def has_dag(cls, fileloc: str, session=None) -> bool:
         :param session: ORM Session
         """
         fileloc_hash = cls.dag_fileloc_hash(fileloc)
-        return session.query(exists().where(cls.fileloc_hash == fileloc_hash)).scalar()
+        return session.query(literal(True)).filter(cls.fileloc_hash == fileloc_hash).first() is not None
```

Comment:
Why this change? Does SQLA not produce the right SQL for MSSQL in this query?

Either way:

```suggestion
        return session.query(literal(True)).filter(cls.fileloc_hash == fileloc_hash).one_or_none() is not None
```

Reply:
yea so this generates an MSSQL query 

```
SELECT EXISTS (SELECT * 
      FROM dag_code        WHERE dag_code.fileloc_hash = ?) AS anon_1
```
which fails  as MSSQL  doesn't support exists  expression in select clause 

The other option was to use 
```
session.query(literal(True)).filter(
            session.query(cls).filter(cls.fileloc_hash == fileloc_hash).exists()
        ).scalar()
```
which worked  on mssql and postgres but not on MSSQL (this MySQL bit is  fixed in a later version of SQLA) 


https://github.com/sqlalchemy/sqlalchemy/issues/5481

Reply:
Ah, and we can't update to SQLA 1.4.o yet because of FAB and sqlalchemy_utils :(

Reply:
@ashb , so we will only have one row per fileloc_hash in `dag_code` ?